CHANGELOG.md

```diff
@@ -48,6 +48,8 @@ different versioning scheme, following the Haskell community's
 * gRPC v1.7.1 is now required to use Bond-over-gRPC.
 * Fixed includes for gRPC services with events or parameterless methods.
   [Issue #735](https://github.com/Microsoft/bond/issues/735)
+* Fixed a bug which would read unrelated struct's field(s) when deserializing base struct
```

Comment:
Minor: a few words missing "...would read an unrelated struct's... deserializing a base struct." and move the period to the end of this sentence have have the "Issue #742" float at the end.